services_app/src/main/java/org/opendatakit/services/preferences/fragments/ServerSettingsFragment.java

```diff
@@ -273,25 +305,172 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
     PropertiesSingleton props = ((IOdkAppPropertiesActivity) this.getActivity()).getProps();
     preference.setSummary((CharSequence) newValue);
     if ( props.containsKey(preference.getKey())) {
-      Map<String,String> properties = new HashMap<String,String>();
-      properties.put(preference.getKey(), newValue.toString());
-      properties.put(CommonToolProperties.KEY_DEFAULT_GROUP, "");
-      properties.put(CommonToolProperties.KEY_ROLES_LIST, "");
-      properties.put(CommonToolProperties.KEY_USERS_LIST, "");
-      props.setProperties(properties);
+      updatePreferenceSummary(preference.getKey(), newValue.toString());
     } else {
       throw new IllegalStateException("Unexpected case");
     }
     return true;
   }
 
+  /**
+   * Receives result after QrCode scanning.
+   */
```

Comment:
Please add @Override

Reply:
Done the required changes. Thanks for suggestions!  @clarlars